src/main/java/htsjdk/samtools/cram/structure/CramCompressionRecord.java

```diff
@@ -133,6 +134,14 @@ private boolean deepEquals(final Collection<?> c1, final Collection<?> c2) {
         return (c1 == null || c1.isEmpty()) && (c2 == null || c2.isEmpty()) || c1 != null && c1.equals(c2);
     }
 
+    @Override
```

Comment:
This is another class with a strange `equals()` method which should be carefully checked.

Reply:
I think this is OK too. The `equals` method will return true only if all of the fields listed here in the `hashCode` method are equal. I think the second part of the contract is actually satisfied in this case too.

Reply:
Here's a violation: `a.readFeatures == null` and `b.readFeatures.isEmpty()`: the `deepEquals()` method means `equals()` is true, but `Objects.hash()` assigns 0 to null and 1 to empty list.

Reply:
Ditto. Fixed and added a test which covers a lot of cases.

Reply:
Looks good.  Would you mind adding a real `readFeatures` list and an empty-string `readName` to the test as well?

Reply:
Done